ci/deps_parser_tests.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import os
+import sys
+import unittest
+from deps_parser import VarImpl
+
+SCRIPT_DIR = os.path.dirname(sys.argv[0])
+CHECKOUT_ROOT = os.path.realpath(os.path.join(SCRIPT_DIR, '..'))
+DEPS = os.path.join(CHECKOUT_ROOT, 'DEPS')
+UPSTREAM_PREFIX = 'upstream_'
+
+
+class TestDepsParserMethods(unittest.TestCase):
+
+  # extract both mirrored dep names and URLs &
+  # upstream names and URLs from DEPs file
+  def setUp(self):
+    with open(DEPS) as file:
+      local_scope_upstream = {}
+      global_scope_upstream = {'Var': lambda x: x}  # dummy lambda
+      # Read the content.
+      with open(DEPS, 'r') as file:
+        deps_content = file.read()
+
+      # Eval the content.
+      exec(deps_content, global_scope_upstream, local_scope_upstream)
+
+      # Extract the upstream URLs
+      # vars contains more than just upstream URLs
+      # however the upstream URLs are prefixed with 'upstream_'
+      upstream = local_scope_upstream.get('vars')
+      self.upstream_urls = upstream
+
+      local_scope_mirror = {}
+      var = VarImpl(local_scope_mirror)
+      global_scope_mirror = {
+          'Var': var.lookup,
+          'deps_os': {},
+      }
+
+      # Eval the content.
+      exec(deps_content, global_scope_mirror, local_scope_mirror)
```

Comment:
I'm not sure why this needs to be done twice. Could the code from lines 36-37 be moved after here (using `local_scope_mirror` instead of `local_scope_upstream`), and the first invocation of `eval` removed?

Reply:
Is this resolved?